### PR TITLE
Split \[Node(Alpha|Serial)Feature:.+\] tags into the proper new tags

### PR DESCRIPTION
**Pull request by mmiranda96**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
As part of #94289, this PR splits test tags from format `\[Node(Alpha|Special)?Feature:.+\]` to `\[Node(Alpha|Special)?Feature\] \[Feature:.+\]`.

#### Which issue(s) this PR fixes:
Partially addresses #94289.

#### Special notes for your reviewer:
This PR should not be merged until https://github.com/kubernetes/test-infra/pull/22677 is merged.
/hold
#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by mmiranda96** [comment]
There's an issue with sysctl test, since it contains `Feature` tag (which by definition is non-conformance). I'll address this particular test case in another PR before merging this one.

**Comment by mmiranda96** [comment]
It looks like there's already some WIP with demoting sysctl test conformance status in #101190. Waiting on it.

File: test/e2e/auth/service_accounts.go
Comment (SergeyKanzhelev) [inline test/e2e/auth/service_accounts.go:488]: we don't need `NodeFeature`, do we? `Feature` + `sig-node` should be sufficient, right? issue mentions that we need to highlight "alpha" features, but it is separate.

Comment (SergeyKanzhelev) [reply test/e2e/auth/service_accounts.go:488]: discussed offline. Need to come up with the plan how to mark these tests as this is not `sig-node` tests. Not clear what the `NodeFeature` tag will signify. It doesn't require any special test config or anything, I think

**Comment by mmiranda96** [comment]
Removing block since https://github.com/kubernetes/kubernetes/pull/103827 has been merged.
/unhold

File: test/e2e_node/hugepages_test.go
Comment (ehashman) [inline test/e2e_node/hugepages_test.go:192]: What makes it special? We need to be careful with this one because if we don't exclude `NodeSpecialFeature` this will get run with the regular `[Feature:.*]` selected tests.

**Review by ehashman — COMMENTED** [review]
Given https://github.com/kubernetes/test-infra/pull/22677 I am comfortable with merging this knowing the tests will still run. Let me kick off a round. One comment below.

**Comment by ehashman** [comment]
/test pull-kubernetes-node-e2e-alpha
/test pull-kubernetes-node-kubelet-serial
/test pull-kubernetes-node-crio-cgrpv2-e2e
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv1
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2
/test pull-kubernetes-node-crio-e2e
/test pull-kubernetes-node-swap-ubuntu
/test pull-kubernetes-node-swap-fedora

these are all the tests changed in https://github.com/kubernetes/test-infra/pull/22677 let's check testgrid to ensure the running suites match 1-1 before merging.

**Comment by SergeyKanzhelev** [comment]
> these are all the tests changed in [kubernetes/test-infra#22677](https://github.com/kubernetes/test-infra/pull/22677) let's check testgrid to ensure the running suites match 1-1 before merging.

@mmiranda96 can you please review failures and confirm that sets of tests running matches what it is before the PR?

**Comment by mmiranda96** [comment]
I'm confirming if all tests are the same. Will update when I have a result.

**Comment by mmiranda96** [comment]
/retest
To compare fresh runs.

**Comment by mmiranda96** [comment]
Some of the tests have timeouts. Rerunning once more...
/test pull-kubernetes-node-e2e-alpha
/test pull-kubernetes-node-kubelet-serial
/test pull-kubernetes-node-crio-cgrpv2-e2e
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv1
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2
/test pull-kubernetes-node-crio-e2e
/test pull-kubernetes-node-swap-ubuntu
/test pull-kubernetes-node-swap-fedora

**Comment by mmiranda96** [comment]
Apologies for the delay. I've uploaded the comparison of ran tests for each of the jobs [here](https://docs.google.com/spreadsheets/d/16A76hjakIoml088Kfp6M9OevNVlUmIDhqbHSVyBapis). The comparison criteria is the following:
- If the old test name includes a tag in the `\[Node[Alpha|Special]?Feature:(.+)\]`, the new test must include a tag `\[Feature:(.+)\], where the capture groups are the same.
- If no tag is found, both names must match exactly.

There are some exceptions:
- `pull-kubernetes-node-e2e-alpha`:  All tests appear as `BeforeSuite`. Numbers match, but there's not a lot of information we can retrieve from these.
- `pull-kubernetes-node-kubelet-serial`: Test keeps timing out.
- `pull-kubernetes-node-swap-ubuntu`: There are no previous runs to compare.
- `pull-kubernetes-node-swap-fedora`: There are some differences, these include tag `[NodeConformance]`.

**Comment by mmiranda96** [comment]
Created https://github.com/kubernetes/kubernetes/issues/104556 to keep track of `pull-kubernetes-node-e2e-alpha` failures.

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by dims** [comment]
/assign @liggitt 

(for sig-auth related approval)

File: test/e2e/auth/service_accounts.go
Comment (liggitt) [reply test/e2e/auth/service_accounts.go:488]: /hold

adding feature tags means these tests will no longer run in default e2e runs, which is unexpected

This test is currently running in presubmit:
https://testgrid.k8s.io/google-gce#pull-kubernetes-e2e-gce-ubuntu-containerd&width=20&include-filter-by-regex=RunAsUser%20or%20FsGroup

But adding feature tags meant it got skipped in presubmit for this PR: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103674/pull-kubernetes-e2e-gce-ubuntu-containerd/1418613003800023040/

**Comment by liggitt** [comment]
`[Feature:...]` tags have special meaning (they are excluded from default e2e runs because they indicate non-default or special config is required). Adding them to all of these tests drops test coverage.

**Comment by ehashman** [comment]
Given @liggitt's comment, can we review anything here that we're adding a `[Feature:..]` tag to and confirm if it's in alpha? If it's beta and beyond, we shouldn't do this, but it's fine for alpha features to have this tag.

**Comment by mmiranda96** [comment]
In beta/GA cases, would it make sense to just have the `\[Node(Alpha|Special}?Feature\]` tag without a `\[Feature:.+\]` tag? If not, what would be a good alternative? Ideally we want to remove all invalid tags, so ignoring those cases isn't an option.

**Comment by ehashman** [comment]
> In beta/GA cases, would it make sense to just have the `\[Node(Alpha|Special}?Feature\]` tag without a `\[Feature:.+\]` tag? If not, what would be a good alternative? Ideally we want to remove all invalid tags, so ignoring those cases isn't an option.

Node(Special)?Feature seems fine, but obviously they are not Alpha if they're beta or GA so we should update them to NodeFeature :)

File: test/e2e_node/hugepages_test.go
Comment (mmiranda96) [reply test/e2e_node/hugepages_test.go:192]: This one already has a `Feature` tag, I'm just replacing the `NodeSpecialFeature` one.

**Comment by mmiranda96** [comment]
I just updated the tags. All fall under one of the following cases:
- If the test previously included a `Feature` tag, `NodeFeature` and `Feature` tags are used.
- If the test has `NodeAlphaFeature`, it splits into `NodeAlphaFeature` and `Feature`.
- Otherwise, `Feature` is not included.

**Comment by dims** [comment]
this PR has been on hold for about 2 weeks. Can you please see how to unblock this by reviewing comments etc? or please close this out (Trying to declutter our open list of PRs)

**Comment by mmiranda96** [comment]
It looks like @SergeyKanzhelev has proposed a KEP related to this (https://github.com/kubernetes/enhancements/issues/3041). Some of the changes from this PR might not align entirely with it.

I think it's OK if I close this and I follow up with a new PR (or a series of PRs) which will be more focused on the KEP itself.

Thanks for bringing this back up @dims!
/close

**Pull request closed without merging**
